src/node/lattice.rs

```diff
@@ -188,90 +191,81 @@ impl ExecutionLattice {
         let mut run_queue = self.run_queue.lock().await;
 
         for event in events {
-            // Iterate through all the roots, and figure out where to add a dependency edge.
-            let mut dfs = DfsPostOrder::empty(&*forest);
-            let mut dependencies: HashSet<NodeIndex<u32>> = HashSet::new();
-            let mut add_edges_to: Vec<NodeIndex<u32>> = Vec::new();
-            let mut add_edges_from: Vec<NodeIndex<u32>> = Vec::new();
-            let mut remove_roots: Vec<RunnableEvent> = Vec::new();
-
-            for root in roots.iter() {
-                // Retrieve the index into the lattice.
-                let root_index: NodeIndex<u32> = root.node_index;
-
-                // This function maintains the stack when iterating over the graph so we do not have to
-                // check the same node in the graph twice.
-                dfs.move_to(root_index);
-
-                // Retrieve the next node in DFS order from the forest.
-                while let Some(nx) = dfs.next(&*forest) {
-                    // Check if any of the children of the current node in DFS traversal already have a
-                    // dependency on the node we are trying to add.
-                    let mut dependent_child: bool = false;
-                    for child in forest.neighbors(nx) {
-                        if dependencies.contains(&child) {
-                            dependencies.insert(nx);
-                            dependent_child = true;
-                            break;
+            // Visits each node in the graph once.
```

Comment:
Can we add a comment here about why we went through a reversal of the forest instead of the way it was originally?